src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -235,18 +227,16 @@ void goto_convertt::remove_post(
 
   // we have ...(op++)...
 
-  if(expr.operands().size()!=1)
-  {
-    error().source_location=expr.find_source_location();
-    error() << "postincrement/postdecrement must have one operand"
-            << eom;
-    throw 0;
-  }
+  DATA_INVARIANT(
+    expr.operands().size() == 1,
+    expr.find_source_location().as_string() +
+      ": postincrement/postdecrement must have one operand");
 
   const irep_idt statement=expr.get_statement();
 
-  assert(statement==ID_postincrement ||
-         statement==ID_postdecrement);
+  DATA_INVARIANT(
+    statement == ID_postincrement || statement == ID_postdecrement,
+    " expected postincrement or postdecrement");
```

Comment:
Here's also a spurious space at the beginning of the error message, which I have gone ahead and removed.